src/domain/account/account-repository.service.ts

```diff
@@ -199,33 +221,36 @@ export class AccountRepository implements IAccountRepository {
     emailAddress: string;
     signer: string;
   }): Promise<void> {
+    const account = await this.accountDataSource.getAccount(args);
     const newEmail = new EmailAddress(args.emailAddress);
-    const currentAccount = await this.accountDataSource.getAccount(args);
 
-    if (newEmail.value === currentAccount.emailAddress.value) {
+    if (newEmail.value === account.emailAddress.value) {
       throw new EmailEditMatchesError(args);
     }
 
-    const verificationCode = this._generateCode();
+    const newVerificationCode = this._generateCode();
 
     await this.accountDataSource.updateAccountEmail({
       chainId: args.chainId,
-      code: verificationCode,
+      code: newVerificationCode,
```

Comment:
This `newVerificationCode` is not being updated in https://github.com/safe-global/safe-client-gateway/blob/5e7817f74560efba5ad02c5a2b2f33d017e6baba/src/datasources/account/account.datasource.ts#L277 or am I missing something?

Related to this, just for consideration: should we have an specific function to update the verification code instead of updating it through `updateAccountEmail`?

Reply:
> This `newVerificationCode` is not being updated in
> 
> https://github.com/safe-global/safe-client-gateway/blob/5e7817f74560efba5ad02c5a2b2f33d017e6baba/src/datasources/account/account.datasource.ts#L277
> 
> or am I missing something?

Good catch! Will fix it.


> Related to this, just for consideration: should we have an specific function to update the verification code instead of updating it through `updateAccountEmail`?

That functionality is already provided with `AccountDataSource.setEmailVerificationCode`.

Reply:
Fixed in [1029720](https://github.com/safe-global/safe-client-gateway/pull/1101/commits/1029720c35fe0377c02abd0e9d0abe197d2ae609).

Added relevant assertions. Also added a `TODO` with a proposed change to tackle in a separate PR as it was not really within the context of this change.